jericho/jericho.py

```diff
@@ -369,36 +375,35 @@ def __init__(self, story_file, seed=-1):
         self.seed = seed
         self.frotz_lib.setup(self.story_file, self.seed)
         self.player_obj_num = self.frotz_lib.get_self_object_num()
+        md5hash = hashlib.md5(open(self.story_file, 'rb').read()).hexdigest()
+        self.bindings = _load_bindings(md5hash)
+        self.act_gen = TemplateActionGenerator(self.bindings) if self.bindings else None
 
     def __del__(self):
         if hasattr(self, 'frotz_lib'):
             self.frotz_lib.shutdown()
             dlclose_func(self.frotz_lib._handle)
 
-    def get_dictionary(self):
-        ''' Returns a list of :class:`jericho.DictionaryWord` words recognized\
-        by the game's parser. See :doc:`dictionary`. '''
-        word_count = self.frotz_lib.get_dictionary_word_count(self.story_file)
-        words = (DictionaryWord * word_count)()
-        self.frotz_lib.get_dictionary(words, word_count)
-        self.frotz_lib.ztools_cleanup()
-        return list(words)
-
-    def disassemble_game(self):
-        ''' Prints the subroutines and strings used by the game. '''
-        self.frotz_lib.disassemble(self.story_file)
-
-    def victory(self):
-        ''' Returns `True` if the game is over and the player has won. '''
-        return self.frotz_lib.victory() > 0
+    def reset(self, use_walkthrough_seed=False):
+        '''
+        Resets the game.
 
-    def game_over(self):
-        ''' Returns `True` if the game is over and the player has lost. '''
-        return self.frotz_lib.game_over() > 0
+        :param use_walkthrough_seed: Seed the emulator to reproduce the walkthrough.
+        :type use_walkthrough_seed: Boolean
+        :returns: A tuple containing the initial observation,\
+        and a dictionary of info.
+        :rtype: string, dictionary
 
-    def emulator_halted(self):
-        ''' Returns `True` if the emulator has halted. To fix a halted game, use :meth:`jericho.FrotzEnv.reset`. '''
-        return self.frotz_lib.halted() > 0
+        '''
+        self.close()
+        seed = self.seed
+        if not self.is_fully_supported or not self.bindings:
+            warnings.warn('Unable to find walkthrough seed for this game.', UnsupportedGameWarning)
```

Comment:
@mhauskn I think the warning message here is misleading. I don't think it should be about "walkthrough". Especially, a warning is raised explicitly in `get_walkthrough`.

Reply:
In addition, another warning is raised when creating a new instance of `FrotzEnv` when dealing with unsupported games.

Reply:
Good point. This warning should only be thrown if use_walkthrough_seed=True. I've updated the code. Let me know if this address the comment.

Reply:
Yes. That's better.